[JSC] Fix successor block processing in Air generation

syg opened the pull request:
#### 4d1ec48c2922efc721720ebb423674d8301c3648
<pre>
[JSC] Fix successor block processing in Air generation
<a href="https://bugs.webkit.org/show_bug.cgi?id=303574">https://bugs.webkit.org/show_bug.cgi?id=303574</a>
<a href="https://rdar.apple.com/165279920">rdar://165279920</a>

Reviewed by Yusuke Suzuki.

The new Air block ordering optimization algorithm introduced in 302875@main
misses processing entrypoints after the first one in the case when the first
entrypoint doesn&apos;t itself cause any successors to be appended.

Unfortunately the test is an infinite loop and has to be skipped until watchdog
works in Wasm.

Test: JSTests/wasm/stress/b3-opt-block-order-multiple-entrypoints.js
Canonical link: <a href="https://commits.webkit.org/303954@main">https://commits.webkit.org/303954@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7605b3dd8c4674acf697fed1f8d75004e2c1101f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/134132 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/6643 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/45340 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/141712 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/86194 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/6664fa30-4a1b-4803-af4f-7b8310ec490e) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/7178 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/6507 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/102576 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/86194 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/8f994ccd-b3a1-4702-b49e-4630be65c914) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/137079 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/5007 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/120232 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/83371 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/c68245a8-f945-42e9-b3a4-78a42e3ea963) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/4887 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/2521 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🛠 wpe-cairo ](https://ews-build.webkit.org/#/builders/WPE-Cairo-Build-EWS "Waiting in queue, processing has not started yet") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/126209 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/114058 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/38361 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/144358 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/132646 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/6313 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/38940 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/110949 "Build is in progress. Recent messages:") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/6395 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/5319 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/111189 "Found 1 new API test failure: WebKitGTK/TestWebsiteData:/webkit/WebKitWebsiteData/itp (failure)") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28189 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/4753 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/116499 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/60083 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/6365 "Built successfully") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/34724 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/165609 "Built successfully") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/157/builds/6211 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/69831 "Built successfully") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/43254 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/6456 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/6319 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
EWS run on current version of this PR (hash https://github.com/WebKit/WebKit/commit/7605b3dd8c4674acf697fed1f8d75004e2c1101f)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/134132 "Passed style check") | [⏳ 🛠 ios ](https://ews-build.webkit.org/#/builders/iOS-26-Build-EWS "Waiting in queue, processing has not started yet") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/45340 "Build is in progress. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | [⏳ 🛠 wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 win ](https://ews-build.webkit.org/#/builders/Win-Build-EWS "Waiting in queue, processing has not started yet") | ⏳ 🛠 ios-apple 
| | [⏳ 🛠 ios-sim ](https://ews-build.webkit.org/#/builders/iOS-26-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 mac-AS-debug ](https://ews-build.webkit.org/#/builders/macOS-Tahoe-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 wpe-wk2 ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 win-tests ](https://ews-build.webkit.org/#/builders/Win-Build-EWS "Waiting in queue, processing has not started yet") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/137079 "Passed tests") | [⏳ 🧪 ios-wk2 ](https://ews-build.webkit.org/#/builders/iOS-26-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-mac](https://ews-build.webkit.org/#/builders/138/builds/45340 "Build is in progress. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | [⏳ 🧪 api-wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | | ⏳ 🛠 vision-apple 
| | [⏳ 🧪 ios-wk2-wpt ](https://ews-build.webkit.org/#/builders/iOS-26-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/138/builds/45340 "Build is in progress. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | [⏳ 🛠 wpe-cairo ](https://ews-build.webkit.org/#/builders/WPE-Cairo-Build-EWS "Waiting in queue, processing has not started yet") | | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/126209 "Build is in progress. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled JSC (warnings); Running jscore-test") | [⏳ 🧪 api-ios ](https://ews-build.webkit.org/#/builders/iOS-26-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/138/builds/45340 "Build is in progress. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/144358 "Built successfully") | | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/132646 "Build is in progress. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled JSC (warnings); Running jscore-test") | [⏳ 🛠 vision ](https://ews-build.webkit.org/#/builders/visionOS-26-Build-EWS "Waiting in queue, processing has not started yet") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/138/builds/45340 "Build is in progress. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | | 
| | [⏳ 🛠 vision-sim ](https://ews-build.webkit.org/#/builders/visionOS-26-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-AS-debug-wk2 ](https://ews-build.webkit.org/#/builders/macOS-Tahoe-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/111189 "Build is in progress. Recent messages:") | | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-26-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/138/builds/45340 "Build is in progress. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | [⏳ 🛠 playstation ](https://ews-build.webkit.org/#/builders/PlayStation-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [⏳ 🛠 tv ](https://ews-build.webkit.org/#/builders/tvOS-26-Build-EWS "Waiting in queue, processing has not started yet") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/138/builds/45340 "Build is in progress. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/165609 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-jsc") | | 
| | [⏳ 🛠 tv-sim ](https://ews-build.webkit.org/#/builders/tvOS-26-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 mac-safer-cpp ](https://ews-build.webkit.org/#/builders/macOS-Safer-CPP-Checks-EWS "Waiting in queue, processing has not started yet") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/35/builds/165609 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-jsc") | | 
| | [⏳ 🛠 watch ](https://ews-build.webkit.org/#/builders/watchOS-26-Build-EWS "Waiting in queue, processing has not started yet") | | | | 
| | [⏳ 🛠 watch-sim ](https://ews-build.webkit.org/#/builders/watchOS-26-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | | | </details>

syg commented on JSTests/wasm/stress/b3-opt-block-order-multiple-entrypoints.js:
```diff
@@ -0,0 +1,72 @@
+
+function f22() {
+}
+const v112 = new WebAssembly.Instance(new WebAssembly.Module(new Uint8Array([
```
It's a fuzzer-generated case and not very illuminating unfortunately. I'll reproduce it here for posterity:

```
(module
 (type $0 (func (result f64 nullref nullexnref)))
 (type $1 (func (param i32) (result i32)))
 (type $2 (func (result externref)))
 (type $3 (func))
 (import "imports" "import_0_v22" (func $fimport$0 (result externref)))
 (table $0 3 10 funcref)
 (elem $0 (i32.const 0) $0 $1 $2)
 (export "iw0" (func $fimport$0))
 (export "w0" (func $0))
 (export "w1" (func $1))
 (export "w2" (func $2))
 (export "w3" (func $3))
 (export "wt0" (table $0))
 (func $0 (param $0 i32) (result i32)
  (local $1 f64)
  (local $2 i32)
  (local $3 f64)
  (local $4 f32)
  (local $5 i32)
  (local $6 i64)
  (local $7 f32)
  (local $8 f32)
  (local.set $1
   (f64.const -536870912)
  )
  (local.set $2
   (i32.trunc_f64_s
    (local.get $1)
   )
  )
  (return
   (local.get $0)
  )
  (local.set $3
   (f64.neg
    (local.get $1)
   )
  )
  (local.set $4
   (f32.const 98)
  )
  (local.set $5
   (i32.trunc_f32_u
    (local.get $4)
   )
  )
  (local.set $6
   (i64.const 2)
  )
  (local.set $7
   (f32.const -227.64126586914062)
  )
  (local.set $8
   (f32.const -105951.21875)
  )
  (local.get $0)
 )
 (func $1 (param $0 i32) (result i32)
  (local $1 (ref i31))
  (local $2 i32)
  (local $3 i32)
  (local $4 f64)
  (local $5 i32)
  (local $6 i32)
  (local $7 i32)
  (local $8 i32)
  (local $9 nullref)
  (local $10 nullexnref)
  (local $11 nullexnref)
  (local $12 nullref)
  (local $13 f64)
  (local $scratch i32)
  (local $scratch_15 (tuple f64 nullref nullexnref))
  (local $scratch_16 nullref)
  (local $scratch_17 f64)
  (local.set $1
   (ref.i31
    (local.get $0)
   )
  )
  (local.set $2
   (i32.const 0)
  )
  (local.set $scratch
   (local.get $2)
  )
  (local.set $13
   (block (result f64)
    (local.set $scratch_17
     (tuple.extract 3 0
      (local.tee $scratch_15
       (loop $label0 (type $0) (result f64 nullref nullexnref)
        (block $label1 (type $0) (result f64 nullref nullexnref)
         (local.set $scratch
          (block $label (result i32)
           (br $label1
            (block (type $0) (result f64 nullref nullexnref)
             (local.set $3
              (local.get $scratch)
             )
             (local.set $4
              (f64.convert_i32_s
               (local.get $3)
              )
             )
             (return
              (local.get $0)
             )
             (local.set $5
              (i32.const 1)
             )
             (local.set $6
              (i32.add
               (local.get $2)
               (local.get $5)
              )
             )
             (local.set $7
              (i32.const 7)
             )
             (local.set $8
              (i32.lt_s
               (local.get $6)
               (local.get $7)
              )
             )
             (drop
              (br_if $label
               (local.get $6)
               (local.get $8)
              )
             )
             (local.set $9
              (ref.null none)
             )
             (local.set $10
              (ref.null noexn)
             )
             (tuple.make 3
              (local.get $4)
              (local.get $9)
              (local.get $10)
             )
            )
           )
          )
         )
         (br $label0)
        )
       )
      )
     )
    )
    (local.set $12
     (block (result nullref)
      (local.set $scratch_16
       (tuple.extract 3 1
        (local.get $scratch_15)
       )
      )
      (local.set $11
       (tuple.extract 3 2
        (local.get $scratch_15)
       )
      )
      (local.get $scratch_16)
     )
    )
    (local.get $scratch_17)
   )
  )
  (local.get $0)
 )
 (func $2 (param $0 i32) (result i32)
  (local $1 f32)
  (local $2 i64)
  (local $3 externref)
  (local $4 i32)
  (local.set $1
   (f32.const 18070)
  )
  (local.set $2
   (i64.trunc_f32_s
    (local.get $1)
   )
  )
  (return_call $0
   (local.get $0)
  )
  (if
   (local.get $0)
   (then
    (local.set $3
     (call $fimport$0)
    )
    (local.set $1
     (local.get $1)
    )
   )
   (else
    (local.set $4
     (i32.le_u
      (local.get $0)
      (local.get $0)
     )
    )
    (local.set $1
     (local.get $1)
    )
   )
  )
  (local.get $0)
 )
 (func $3
  (local $0 i32)
  (local $1 i32)
  (local $2 externref)
  (local $3 i32)
  (local $4 i32)
  (local $5 i32)
  (local $6 i32)
  (local $7 f64)
  (local $8 i32)
  (local $9 i32)
  (local $10 i32)
  (local $11 i32)
  (local $12 i32)
  (local $13 f64)
  (local $14 externref)
  (local $15 externref)
  (local $16 i32)
  (local $17 externref)
  (local $18 (ref i31))
  (local $19 i32)
  (local $20 i32)
  (local $scratch i32)
  (local $scratch_22 (tuple i32 i32 i32 i32))
  (local $scratch_23 (tuple i32 i32 i32 i32))
  (local $scratch_24 i32)
  (local $scratch_25 i32)
  (local $scratch_26 i32)
  (local.set $0
   (i32.const 0)
  )
  (local.set $scratch
   (local.get $0)
  )
  (loop $label0
   (block $label1
    (local.set $scratch
     (block $label (result i32)
      (local.set $1
       (local.get $scratch)
      )
      (drop
       (br_if $label
        (local.get $0)
        (local.get $1)
       )
      )
      (drop
       (br $label
        (local.get $0)
       )
      )
      (local.set $2
       (call $fimport$0)
      )
      (local.set $3
       (i32.const 1)
      )
      (local.set $4
       (i32.add
        (local.get $0)
        (local.get $3)
       )
      )
      (local.set $5
       (i32.const 12)
      )
      (local.set $6
       (i32.lt_s
        (local.get $4)
        (local.get $5)
       )
      )
      (drop
       (@metadata.code.branch_hint "\01")
       (br_if $label
        (local.get $4)
        (local.get $6)
       )
      )
      (br $label1)
     )
    )
    (br $label0)
   )
  )
  (local.set $7
   (f64.convert_i32_u
    (local.get $0)
   )
  )
  (local.set $8
   (i32.ge_s
    (local.get $0)
    (local.get $0)
   )
  )
  (local.set $scratch_22
   (tuple.make 4
    (local.get $0)
    (local.get $0)
    (local.get $0)
    (local.get $8)
   )
  )
  (try
   (do
    (local.set $12
     (block (result i32)
      (local.set $scratch_26
       (tuple.extract 4 0
        (local.tee $scratch_23
         (local.get $scratch_22)
        )
       )
      )
      (local.set $11
       (block (result i32)
        (local.set $scratch_25
         (tuple.extract 4 1
          (local.get $scratch_23)
         )
        )
        (local.set $10
         (block (result i32)
          (local.set $scratch_24
           (tuple.extract 4 2
            (local.get $scratch_23)
           )
          )
          (local.set $9
           (tuple.extract 4 3
            (local.get $scratch_23)
           )
          )
          (local.get $scratch_24)
         )
        )
        (local.get $scratch_25)
       )
      )
      (local.get $scratch_26)
     )
    )
    (local.set $13
     (select
      (local.get $7)
      (local.get $7)
      (local.get $11)
     )
    )
    (local.set $14
     (call $fimport$0)
    )
    (local.set $15
     (call $fimport$0)
    )
   )
   (catch_all
    (local.set $16
     (call $1
      (local.get $0)
     )
    )
    (local.set $17
     (call $fimport$0)
    )
    (local.set $18
     (ref.i31
      (local.get $16)
     )
    )
   )
  )
  (local.set $19
   (i32.const 1)
  )
  (local.set $20
   (call $0
    (local.get $19)
   )
  )
 )
)

```

syg committed:
[JSC] Fix successor block processing in Air generation
https://bugs.webkit.org/show_bug.cgi?id=303574
rdar://165279920

Reviewed by Yusuke Suzuki.

The new Air block ordering optimization algorithm introduced in 302875@main
misses processing entrypoints after the first one in the case when the first
entrypoint doesn't itself cause any successors to be appended.

Unfortunately the test is an infinite loop and has to be skipped until watchdog
works in Wasm.

Test: JSTests/wasm/stress/b3-opt-block-order-multiple-entrypoints.js
Canonical link: https://commits.webkit.org/303954@main
